syntax = "proto3";

// Each topic in the response array should have a keywords list and a correlation probability
message SingleTopic{
    repeated string keywords=1;
    float correlation=2;
}

// List of SingleTopics
message AllTopics{
    repeated SingleTopic topics=1;
}

// Request needs a query description
message PredictTopicRequest{
    string query_description=1;
}

// Response will be a list of tuples where each tuple is a SingleTopic with keywords list and correlation probability
message PredictTopicResponse{
    AllTopics topic_distribution=1;
}

// Define a service with the appropriate input and output
service PredictLda{
    rpc predict(PredictTopicRequest) returns (PredictTopicResponse);
}